internal/service/fw/api_internal_domain_lists_data_source.go

```diff
@@ -0,0 +1,115 @@
+package fw
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	bloxoneclient "github.com/infobloxopen/bloxone-go-client/client"
+	"github.com/infobloxopen/bloxone-go-client/fw"
+	"github.com/infobloxopen/terraform-provider-bloxone/internal/flex"
+	"github.com/infobloxopen/terraform-provider-bloxone/internal/utils"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ datasource.DataSource = &InternalDomainListsDataSource{}
+
+func NewInternalDomainListsDataSource() datasource.DataSource {
+	return &InternalDomainListsDataSource{}
+}
+
+// InternalDomainListsDataSource defines the data source implementation.
+type InternalDomainListsDataSource struct {
+	client *bloxoneclient.APIClient
+}
+
+func (d *InternalDomainListsDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_" + "td_internal_domain_lists"
+}
+
+type ModelWithFilter struct {
+	Filters    types.Map  `tfsdk:"filters"`
+	TagFilters types.Map  `tfsdk:"tag_filters"`
+	Results    types.List `tfsdk:"results"`
+}
+
+func (m *ModelWithFilter) FlattenResults(ctx context.Context, from []fw.AtcfwInternalDomains, diags *diag.Diagnostics) {
+	if len(from) == 0 {
+		return
+	}
+	m.Results = flex.FlattenFrameworkListNestedBlock(ctx, from, AtcfwInternalDomainsAttrTypes, diags, FlattenAtcfwInternalDomains)
+}
+
+func (d *InternalDomainListsDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "",
```

Comment:
Please add description. (applies to all resources and data sources)

Reply:
Added